lib/jwt_sessions/authorization.rb

```diff
@@ -27,9 +27,13 @@ def authorize_refresh_by_access_request!
       rescue Errors::Unauthorized
         cookie_based_auth(:access)
       end
+
+      ruid = JWTSessions::Session.new.access_token_ruid(found_token)
+      refresh_token = RefreshToken.find(ruid, JWTSessions.token_store)
```

Comment:
please move it to session.rb, only leave safe_check_csrf and raise invalid_authorization from within session.rb